labgrid/pytestplugin/hooks.py

```diff
@@ -110,18 +115,30 @@ def pytest_collection_modifyitems(config, items):
             elif isinstance(arg, list):
                 want_feature.update(arg)
             else:
-                raise Exception("Unsupported feature argument type")
+                raise pytest.UsageError("Unsupported argument type in pytest.mark.lg_feature()")
+
         missing_feature = want_feature - have_feature
         if missing_feature:
-            if len(missing_feature) == 1:
-                skip = pytest.mark.skip(
-                    reason=f'Skipping because feature "{missing_feature}" is not supported'
-                )
-            else:
-                skip = pytest.mark.skip(
-                    reason=f'Skipping because features "{missing_feature}" are not supported'
+            reason = f'unsupported feature(s): {", ".join(missing_feature)}'
+            item.add_marker(pytest.mark.skip(reason=reason))
+
+        # pytest.mark.lg_xfail_feature
+        for marker in item.iter_markers("lg_xfail_feature"):
+            [feature] = marker.args
+            if not isinstance(feature, str):
+                raise pytest.UsageError("Unsupported feature argument type in pytest.mark.lg_xfail_feature")
```

Comment:
Please add the wrong and expected type here, this will make it easier for the user to change the code without going to the source code.